### PR TITLE
Create no data event screen in Saved screen

### PR DESCRIPTION
**Pull request by yongmin86k**
# Create no data event screen in Saved screen

- Add async-storage methods for saved events and self-guided tours
    - Create SaveContext and SaveProvider
    - Add saveId and removeId functionalities in Home screen
    - Add saveId and removeId functionalities in EventCategory screen

- Create segmented controllers for Saved screen
    - Create no data screen for TabEvents screen

image

image

### CODE REVIEW

File: client/js/App.js [hunk 1]
```diff
@@ -2,12 +2,15 @@ import React, {Component} from 'react';
...
+import {SaveProvider} from './context/SaveContext';
 
 export default class App extends Component {
   render() {
     return (
       <ApolloProvider client={client}>
-        <RootStackNavigator />
+        <SaveProvider>
+          <RootStackNavigator />
+        </SaveProvider>
```
Comment (siddharthparmar7) [inline client/js/App.js]: What is `SaveProvider`?

File: client/js/components/CardEvent/CardEvent.js [hunk 2]
```diff
@@ -27,64 +29,88 @@ const CardEvent = ({data}) => {
   const hour = 12;
 
   return (
-    <TouchableOpacity style={styles.container}>
-      <ImageBackground source={{uri: image}} style={styles.bgImg}>
-        <View style={styles.bgImgWrapper}>
-          <Text ellipsizeMode="tail" numberOfLines={1} style={styles.title}>
-            {title}
-          </Text>
+    <SaveContext.Consumer>
```
Comment (siddharthparmar7) [inline client/js/components/CardEvent/CardEvent.js]: What is the purpose of this context?

File: client/js/components/CardEventSmall/CardEventSmall.js [hunk 3]
```diff
@@ -7,45 +7,66 @@ import {
...
-      </Text>
-      <Text style={styles.price} ellipsizeMode="tail" numberOfLines={1}>
-        {price ? `$${price}` : 'Free'}
-      </Text>
-      <View style={styles.wrapperRating}>
-        <RatingScore score={calculateRatingScore(reviews)} />
-        <Text style={styles.ratingText}>({reviews.length})</Text>
-      </View>
-    </TouchableOpacity>
+    <SaveContext.Consumer>
```
Comment (siddharthparmar7) [inline client/js/components/CardEventSmall/CardEventSmall.js]: same as above.

File: client/js/context/SaveContext.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import React, {Component} from 'react';
+import {addSave, removeSave, getAllSaves} from '../config/models';
+
+const SaveContext = React.createContext();
+export default SaveContext;
+
+export class SaveProvider extends Component {
```
Comment (siddharthparmar7) [inline client/js/context/SaveContext.js]: what is the purpose of `SaveProvider`?

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

File: client/js/App.js (hunk 1 above)
Comment (yongmin86k) [reply client/js/App.js]: It's like FaveProvider in r10, but designers named it "Save", so I named it as SaveProvider

it stores the id for saved events and self-guided tours

File: client/js/components/CardEvent/CardEvent.js (hunk 2 above)
Comment (yongmin86k) [reply client/js/components/CardEvent/CardEvent.js]: Same as above

Comment (yongmin86k) [reply client/js/components/CardEvent/CardEvent.js]: Same as above

File: client/js/components/CardEventSmall/CardEventSmall.js (hunk 3 above)
Comment (yongmin86k) [reply client/js/components/CardEventSmall/CardEventSmall.js]: Same as above

File: client/js/context/SaveContext.js (hunk 4 above)
Comment (yongmin86k) [reply client/js/context/SaveContext.js]: Same as above

File: client/js/App.js (hunk 1 above)
Comment (yongmin86k) [reply client/js/App.js]: Fixed

File: client/js/components/CardEvent/CardEvent.js (hunk 2 above)
Comment (yongmin86k) [reply client/js/components/CardEvent/CardEvent.js]: Fixed

File: client/js/components/CardEventSmall/CardEventSmall.js (hunk 3 above)
Comment (yongmin86k) [reply client/js/components/CardEventSmall/CardEventSmall.js]: Fixed

File: client/js/context/SaveContext.js (hunk 4 above)
Comment (yongmin86k) [reply client/js/context/SaveContext.js]: Fixed

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**